File: src/singularity/type/SglCategory.java
package singularity.type;

import mindustry.type.Category;
import singularity.Sgl;
import universecore.UncCore;

import static singularity.Sgl.modName;

public class SglCategory {
  /** 各种生产/传输核能的设备; 消耗核能的工厂应当保持在crafting中 */
  public static Category nuclear;
  /** 矩阵网络相关的主要方块 */
  public static Category matrix;
  /** 调试方块，在配置game debug mod开启时，在沙盒模式可见 */
  public static Category debugging;

  public static void load() {
    nuclear = UncCore.INSTANCE.getCategories().add("nuclear",  modName + "-nuclear");
    matrix = UncCore.INSTANCE.getCategories().add("matrix",  modName + "-matrix");

    if (Sgl.config.debugMode) debugging = UncCore.INSTANCE.getCategories().add("debugging", modName + "-debugging");
  }
}

File: src/singularity/type/SglCategory.kt
package singularity.type

import ice.DeepSpace.modName
import ice.library.world.Load
import mindustry.type.Category
import universecore.UncCore.categories

object SglCategory: Load{
  /** 各种生产/传输核能的设备;消耗核能的工厂应当保持在crafting中  */
  var nuclear: Category = categories.add("nuclear", "$modName-nuclear")
  /** 矩阵网络相关的主要方块  */
  var matrix: Category = categories.add("matrix", "$modName-matrix")
  /** 静态环境方块  */
  var environment: Category = categories.add("environment", "$modName-environment")
}